### PR TITLE
fix(extensions): bypass cors in requests made from background pages

### PR DESCRIPTION
**Pull request by sentialx**
#### Description of Change
This PR fixes CORS issues mentioned in https://github.com/electron/electron/issues/19447#issuecomment-619365573.
Could this be backported to 9-x-y, please?

cc @nornagon 

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] Tests are changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).
- [x] This is **NOT A BREAKING CHANGE**. Breaking changes may not be merged to master until 11-x-y is branched.

#### Release Notes

Notes: Fixed issues with CORS when making requests from extensions.

### CODE REVIEW

**Comment by codebytere** [comment]
@sentialx would it be possible to add a test for this?

**Comment by sentialx** [comment]
@codebytere Are the tests okay?

File: spec-main/extensions-spec.ts [hunk 1]
```diff
@@ -32,6 +32,19 @@ describe('chrome extensions', () => {
...
+    return contents.executeJavaScript(`fetch(${JSON.stringify(url)})`);
+  }
+
+  it('bypasses CORS in requests made from extensions', async () => {
+    const customSession = session.fromPartition(`persist:${require('uuid').v4()}`);
+    const w = new BrowserWindow({ show: false, webPreferences: { session: customSession, sandbox: true } });
+    const { id } = await customSession.loadExtension(path.join(fixtures, 'extensions', 'ui-page'));
+    w.loadURL(`chrome-extension://${id}/bare-page.html`);
+    await emittedOnce(w.webContents, 'dom-ready');
+    await expect(fetch(w.webContents, 'https://google.com')).to.not.be.rejected;
```
Comment (nornagon) [inline spec-main/extensions-spec.ts:45]:
```suggestion
    await expect(fetch(w.webContents, 'https://google.com')).to.not.be.rejected();
```

**Comment by sentialx** [comment]
Oh no, I forgot to add `<all_urls>` to permissions.

**Pull request merged**